### PR TITLE
DW-3874: Add WAF logs

### PR DESCRIPTION
**Pull request by mattburgess**

### CODE REVIEW

File: terraform/deploy/modules.tf [hunk 1]
```diff
@@ -172,12 +172,22 @@ module "vpc" {
...
-module "waf" {
-  source = "../modules/waf"
+module "concourse_waf_log_group" {
+  source = "../modules/cloudwatch_log_group"
 
   name = var.name
+  tags = local.tags
 
+  group_name        = "waf"
+  retention_in_days = 30
```
Comment (dan-hill2802) [inline terraform/deploy/modules.tf:182]: should probably be 180
> For the Management environment: all management logs should be retained for 6 months.

File: terraform/deploy/modules.tf [hunk 2]
```diff
@@ -172,12 +172,22 @@ module "vpc" {
   vpc_endpoint_source_sg_ids  = [module.concourse_web.outputs.security_group.id, module.concourse_worker.outputs.security_group.id]
 }
 
-module "waf" {
-  source = "../modules/waf"
+module "concourse_waf_log_group" {
```
Comment (dan-hill2802) [inline terraform/deploy/modules.tf:175]: wondering if this should simply be a resource. Is the module adding anything?

File: terraform/modules/waf/kinesis.tf [hunk 3]
```diff
@@ -0,0 +1,73 @@
+resource "aws_kinesis_firehose_delivery_stream" "extended_s3_stream" {
+  name        = "aws-waf-logs-${var.name}"
+  destination = "extended_s3"
+
```
Comment (dan-hill2802) [inline terraform/modules/waf/kinesis.tf:4]: can you add common tags please?

File: terraform/modules/waf/kinesis.tf [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    resources = [
+      "${var.log_bucket}"
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
```
Comment (dan-hill2802) [inline terraform/modules/waf/kinesis.tf:63]: will `kms:encrypt`  be required?

File: terraform/deploy/modules.tf (hunk 1 above)
Comment (jimbarker) [reply terraform/deploy/modules.tf:182]: Agree. This should be 180.

Comment (jimbarker) [reply terraform/deploy/modules.tf:182]: Does the Analytics repo need updating to 180 as well?

File: terraform/modules/waf/kinesis.tf [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+data "aws_iam_policy_document" "write_waf_logs" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:ListBucket",
+    ]
+
+    resources = [
+      "${var.log_bucket}"
```
Comment (dan-hill2802) [inline terraform/modules/waf/kinesis.tf:58]:
```suggestion
      var.log_bucket
```

Comment (mattburgess) [reply terraform/modules/waf/kinesis.tf:58]: Uggh, how did that sneak in there. Sorry :) Either @ChrisScottThomas or myself will fix up in the morning.

**Pull request merged**